edith-step/src/App.js

```diff
@@ -1,46 +1,102 @@
 import React, {Component} from 'react';
-import ReceiptInput from './ReceiptInput';
-import LineChart, {BarChart} from './UserChart';
-import axios from 'axios';
-//import "./App.css";
+import LineChart, {BarGraph, CategoryDoughnutChart, ItemDoughnutChart} from './UserChart';
+import ReceiptInput from './components/ReceiptInput';
+import TopNavbar from './components/TopNavbar';
+import './App.css';
 
+/**
+ * Corresponds to the different chart types.
+ */
+ const chart = {
+    'LINE': 'LINE',
+    'BAR': 'BAR',
+    'DOUGHNUT':  'DOUGHNUT' 
+}
 
-/** Main webpage for the website. */
+/**
+ * Main webpage for the website.
+ */
 class App extends Component {
   constructor() {
     super();
-    this.state = { "chartType" : LineChart }; 
-    this.updateChartType = this.updateChartType.bind(this);   
-  }
+    this.state = {'chartType': LineChart, 'dateSelection': '', 'categorySelection': ''}; 
+    this.updateChartType = this.updateChartType.bind(this); 
+    this.showItemChart = this.showItemChart.bind(this);  
+  } 
 
+  /**
+   * Updates the value of chartType in state 
+   * based on the value of {@code event}.
+   * @param event - HTML element that has been selected
+   */
   updateChartType(event) {
     switch(event.target.value) {
-      case "LineChart": 
-        this.setState({"chartType": LineChart});
+      case chart.LINE:
+        this.setState({'chartType': LineChart});
+        break;
+      case chart.BAR:
+        this.setState({'chartType': BarGraph});
         break;
-      case "BarChart":
-        this.setState({"chartType": BarChart});
+      case chart.DOUGHNUT:
+        this.setState({'chartType': CategoryDoughnutChart});
         break;
     }
   }
 
+  /**
+   * Displays either a {@code CategoryDoughnutChart} or
+   * {@code ItemDoughnutChart} based on the value of {@code doughnutType}
+   * @param doughnutType can be eiter 'category' or 'item'
+   * @param dateSelection if this value is not empty, only items/categories 
+   *                      bought in the same week will be displayed
+   * @param categroySelection if this value is not empty, only items of this
```

Comment:
how about somethign like categoryType? or categoryFilter?

Reply:
I changed it to categoryType.